pkg/proxy/iptables/proxier.go

```diff
@@ -1355,16 +1362,24 @@ func (proxier *Proxier) syncProxyRules() {
 			if len(localEndpoints) != 0 {
 				// Write rules jumping from policyLocalChain to localEndpointChains
 				proxier.writeServiceToEndpointRules(svcNameString, svcInfo, policyLocalChain, localEndpoints, args)
-			} else if hasEndpoints {
-				// Blackhole all traffic since there are no local endpoints
-				args = append(args[:0],
-					"-A", string(policyLocalChain),
-					"-m", "comment", "--comment",
-					fmt.Sprintf(`"%s has no local endpoints"`, svcNameString),
-					"-j",
-					string(KubeMarkDropChain),
-				)
-				proxier.natRules.Write(args)
+			} else {
+				if svcInfo.NodeLocalInternal() && utilfeature.DefaultFeatureGate.Enabled(features.ServiceInternalTrafficPolicy) {
+					serviceNoLocalEndpointsTotalInternal++
+				}
+				if svcInfo.NodeLocalExternal() {
+					serviceNoLocalEndpointsTotalExternal++
+				}
+				if hasEndpoints {
+					// Blackhole all traffic since there are no local endpoints
+					args = append(args[:0],
+						"-A", string(policyLocalChain),
+						"-m", "comment", "--comment",
+						fmt.Sprintf(`"%s has no local endpoints"`, svcNameString),
+						"-j",
+						string(KubeMarkDropChain),
+					)
+					proxier.natRules.Write(args)
```

Comment:
hm... that use of args is unnecessary with the current code...

```suggestion
					proxier.natRules.Write(
						"-A", string(policyLocalChain),
						"-m", "comment", "--comment",
						fmt.Sprintf(`"%s has no local endpoints"`, svcNameString),
						"-j",
						string(KubeMarkDropChain),
					)
```

Reply:
I have just moved the code. Is it OK if we keep it as is for this PR? I'm getting an exception since we are past the freeze and my justification for the low risk is that we are simply adding a metric.